Makefile

```diff
@@ -136,11 +144,12 @@ tidy: ## Run go mod tidy
 
 .PHONY: test
 test: manifests generate fmt vet envtest ## Run tests.
-	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" GOFLAGS=$(GOFLAGS) go test ./... -coverprofile cover.out
+	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" GOFLAGS=$(GOFLAGS) go test $(go list ./... | grep -v /test/) -coverprofile cover.out
 
+LABEL_FILTER=gatekeeper-controller
 .PHONY: test-e2e
-test-e2e: generate fmt vet ## Run e2e tests, using the configured Kubernetes cluster in ~/.kube/config
-	GOFLAGS=$(GOFLAGS) USE_EXISTING_CLUSTER=true go test -v ./test/e2e -coverprofile cover.out -race -args -ginkgo.v -ginkgo.progress -ginkgo.trace -namespace $(NAMESPACE) -timeout 5m -delete-timeout 10m
+test-e2e: generate fmt vet install-crds ## Run e2e tests, using the configured Kubernetes cluster in ~/.kube/config
+	GOFLAGS=$(GOFLAGS) USE_EXISTING_CLUSTER=true go test -v ./test/e2e -coverprofile cover.out -race -args -ginkgo.v -ginkgo.trace -ginkgo.label-filter=$(LABEL_FILTER) -ginkgo.fail-fast -namespace $(NAMESPACE) -timeout 5m -delete-timeout 10m
```

Comment:
Progress deprecated

Reply:
why it uses `go test`, no idea. it was there. Is there any big difference between ginkgo directly or  `go test`? 
Why I added this `fail-fast`, to get e2e test result faster and the tests are chained(tests are depended on other test) so there is no point we excluded `fail-fast`

Reply:
Let's change it to `ginkgo` so that it's clearer that's what we're using. Plus, the Ginkgo docs recommend it:
>Under the hood, ginkgo is simply calling go test. While you can run go test instead of the ginkgo CLI, Ginkgo has several capabilities that can only be accessed via ginkgo. We generally recommend users embrace the ginkgo CLI and treat it as a first-class member of their testing toolchain.

(ref: https://onsi.github.io/ginkgo/#mental-model-go-modules-packages-and-tests)

With `fail-fast`, we do get results faster but if there are multiple failures, you find out about them one at a time, so the feedback could be slower overall. What tests are chained? Maybe putting them in an `Ordered` container might work for that?

Reply:
I added in my new e2e test but old e2e test is not. Adding `Ordered` explicitly is a good idea. Let me add it. Also I will change to ginkgo Thanks!